scripts/assign_evidence_to_TADs.py

```diff
@@ -84,65 +79,43 @@ def parse_ev_key(tadkey):
-        chrom, TADkey = buildTADkey(tad_sub)
-        FullTADinfo = TADdictgenes[chrom][TADkey]
+for tad_row in range(len(tad_gwas_df)):
+    snp_info = tad_gwas_df.ix[tad_row, :]
 
-        # Build evidence key
-        e_key = 'chr{}:{}'.format(chrom, TADkey)
+    genes = str(snp_info['gene']).split(',')
+    genes = [x.strip(' ') for x in genes]
 
-        if e_key not in evidence_dict.keys():
-            evidence_dict[e_key] = []
+    # Build the key to lookup TAD in dict and lookup
+    e_key = buildTADkey(snp_info)
 
-        # if the TAD is not empty
-        if len(FullTADinfo) > 0:
+    if e_key not in evidence_dict.keys():
+        evidence_dict[e_key] = []
 
-            # Get the genes in the TAD
-            TAD_genes = FullTADinfo.gene[0:].tolist()
+    tad_index = int(snp_info['TADidx'])
+    tad_subset_info = gene_df.ix[gene_df['TAD_id'] == str(tad_index), :]
+    tad_genes = tad_subset_info['gene_name'].tolist()
 
-            # Subset the evidence data frame to genes in TAD
-            evidence_sub = EVIDENCE.ix[EVIDENCE['gene'].isin(TAD_genes), :]
+    # Subset the evidence dataframe to TAD based genes
+    evidence_sub = evidence_df.ix[evidence_df['gene'].isin(tad_genes), :]
 
-            # Loop over each of the rows
-            TAD_gene_evidence = []
-            for ev in range(0, evidence_sub.shape[0]):
-                gene = evidence_sub['gene'].tolist()[ev]
-                ev_type = evidence_sub['evidence'].tolist()[ev]
-                TAD_gene_evidence.append([gene, ev_type])
+    # Loop over each of the rows
+    TAD_gene_evidence = []
+    for ev in range(0, evidence_sub.shape[0]):
+        gene = evidence_sub['gene'].tolist()[ev]
+        ev_type = evidence_sub['evidence'].tolist()[ev]
+        TAD_gene_evidence.append([gene, ev_type])
 
-            # Add this gene and evidence to the dictionary
-            evidence_dict[e_key].append(TAD_gene_evidence)
+    evidence_dict[e_key].append(TAD_gene_evidence)
 
-####################################
 # Write out results to file
-####################################
-with open(OUT_FH, 'w') as out_fh:
+with open(output_file, 'w') as out_fh:
     tadwriter = csv.writer(out_fh, delimiter='\t')
-    tadwriter.writerow(('ID', 'CHROMOSOME', 'START', 'END', 'UCSC', 'GENE',
-                        'EVIDENCE'))
+    tadwriter.writerow(['gene', 'evidence', 'TAD ID', 'chromosome',
+                        'TAD Start', 'TAD End', 'UCSC'])
     for tadkey in evidence_dict.keys():
-        info = parse_ev_key(tadkey)
-        ID, chrom, start, end, ucsc = info
-        loop_idx = 0
+        ID, chrom, start, end, ucsc = parse_ev_key(tadkey)
         for evidence in evidence_dict[tadkey][0]:
-            gene, e = evidence
-            if loop_idx == 0:
-                tadwriter.writerow((ID, chrom, start, end, ucsc, gene, e))
-            else:
-                tadwriter.writerow([None] * 5 + [gene, e])
-            loop_idx += 1
+            gene, evidence = evidence
```

Comment:
yeah, it is pretty confusing...I'll update the variable name in a new pull request!